tests/Unit/Framework/Container/ContainerTest.php

```diff
@@ -64,17 +66,58 @@ public function test_resolve_service_as_function(): void
 
     public function test_resolve_service_as_callable_class(): void
     {
-        $this->container->set('service_name', new class() {
-            public function __invoke(): string
-            {
-                return 'value';
+        $this->container->set(
+            'service_name',
+            new class() {
+                public function __invoke(): string
+                {
+                    return 'value';
+                }
             }
-        });
+        );
 
         $resolvedService = $this->container->get('service_name');
         self::assertSame('value', $resolvedService);
 
         $cachedResolvedService = $this->container->get('service_name');
         self::assertSame('value', $cachedResolvedService);
     }
+
+    public function test_resolve_non_factory_service_with_random(): void
+    {
+        $this->container->set(
+            'service_name',
+            static fn () => 'value_' . random_int(0, PHP_INT_MAX)
+        );
+
+        self::assertSame(
+            $this->container->get('service_name'),
+            $this->container->get('service_name')
+        );
+    }
+
+    public function test_resolve_factory_service_with_random(): void
+    {
+        $this->container->set(
+            'service_name',
+            $this->container->factory(
```

Comment:
I am not completely convinced of `factory` name.
Probably, I would rename the methods to: 
- `instance` (for the `factory` method)
- `singleton` (for the `get` method)

What do you think? 🤔

Reply:
Good observation, but actually this is the same naming Spryker is using in their [internal Container](https://github.com/spryker/container/blob/master/src/Spryker/Service/Container/Container.php#L587) as well. I kinda like it, and I would prefer to have a similar name in this area.